compiler_opt/es/blackbox_evaluator.py

```diff
@@ -0,0 +1,109 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tooling computing rewards in blackbox learner."""
+
+import abc
+import concurrent.futures
+from typing import List, Optional
+
+from absl import logging
+import gin
+
+from compiler_opt.distributed.worker import FixedWorkerPool
+from compiler_opt.rl import corpus
+from compiler_opt.es import blackbox_optimizers
+from compiler_opt.distributed import buffered_scheduler
+
+
+class BlackboxEvaluator(metaclass=abc.ABCMeta):
+  """Blockbox evaluator abstraction."""
+
+  @abc.abstractmethod
+  def __init__(self, train_corpus: corpus.Corpus):
+    pass
+
+  @abc.abstractmethod
+  def get_results(
+      self, pool: FixedWorkerPool,
+      perturbations: List[bytes]) -> List[concurrent.futures.Future]:
+    raise NotImplementedError()
+
+  @abc.abstractmethod
+  def set_baseline(self) -> None:
+    raise NotImplementedError()
+
+  @abc.abstractmethod
+  def get_rewards(
+      self, results: List[concurrent.futures.Future]) -> List[Optional[float]]:
+    raise NotImplementedError()
+
+
+@gin.configurable
+class SamplingBlackboxEvaluator(BlackboxEvaluator):
```

Comment:
don't you want to also have the non-sampling one implemented, i.e. the old behavior?

Reply:
This is the old behavior. It's basically just copy and pasted from `blackbox_learner.py`.